playwright/fixtures/index.ts

```diff
@@ -35,113 +44,242 @@ export type Fixtures = {
+                {
+                  name: "slice-machine_isInAppGuideOpen",
+                  value: "false",
+                },
+              ],
+            },
+          ],
+        },
+      });
+
+      // Prevent user to be onboarded if needed
+      if (!onboarded) {
+        context = await browser.newContext({
+          storageState: {
+            cookies: [],
+            origins: [
               {
-                name: "slice-machine_isInAppGuideOpen",
-                value: "false",
+                origin: config.use.baseURL,
+                localStorage: [],
               },
             ],
           },
-        ],
-      },
-    });
+        });
+      }
+
+      // Create new page object with new context
+      const page = await context.newPage();
+
+      // Logout user by default
+      try {
+        await fs.rm(path.join(os.homedir(), ".prismic"));
+      } catch (error) {
+        // Ignore since it means the user is already logged out
+      }
+
+      // Login user if needed
+      if (loggedIn) {
+        // In CI we define a PRISMIC_URL env variable to fasten the tests
+        let prismicUrl: string | undefined = process.env["PRISMIC_URL"];
+
+        // In local we get the Prismic URL from the browser, it helps to avoid
+        // switching manually between Wroom and Prismic
+        if (!prismicUrl) {
+          await page.route(
+            "*/**/_manager",
+            async (route) => {
+              const postDataBuffer = route.request().postDataBuffer() as Buffer;
+              const postData = decode(postDataBuffer) as Record<
+                "procedurePath",
+                unknown[]
+              >;
+
+              if (postData.procedurePath[0] === "getState") {
+                const response = await route.fetch();
+                const existingBody = await response.body();
+                const existingData = (
+                  decode(existingBody) as Record<"data", unknown>
+                ).data as Record<
+                  "env",
+                  {
+                    endpoints: { PrismicWroom: string };
+                  }
+                >;
+
+                // Get Prismic URL from the response of the getState call
+                prismicUrl = existingData.env.endpoints.PrismicWroom;
 
-    const page = await context.newPage();
+                await route.continue();
+              }
+            },
+            // Ensure only the first getState call is intercepted
+            {
+              times: 1,
+            },
+          );
 
-    // Use the fixture value in the test.
-    await use(page);
+          // Visit the page to trigger the getState call and wait for it
+          await page.goto("/");
+          await page.waitForResponse("*/**/_manager");
+        }
 
-    // Gracefully close up everything
-    await page.close();
-    await context.close();
-  },
-});
+        const activeEnv = prismicUrl?.includes("wroom") ? "WROOM" : "PRISMIC";
```

Comment:
Determining environments based on URL seems inflexible to me. What do you think of using a dedicated option for setting the environment instead? I'm not sure where that option would go (does Playwright have a way to configure that? Or would it need to be an environment variable?).

Reply:
Ah ^^ The goal was to be flexible. Maybe it's not clear enough but you can already used an environement variable. `PRISMIC_URL`, so all good. 
Why that then? As I just explain above in another comment, it's to improve DX. I did that for some use cases I saw while testing.

I often switch between `yarn run slicemachine:dev` and in another project `npm run slicemachine` so between wroom and prismic. Do I want to update an env variable in Playwright everytime, not really. I just don't wanna care. I wrote the email and password for both env and I want an automatic swtich. 

If you wan to hard set an env you can with `PRISMIC_URL` anyway. That's waht the CI do because you will save 3s per test that need login then and that's cool for CI

Reply:
Discussed together 🙂